Sources/Commands/Workspace.swift

```diff
@@ -504,14 +572,34 @@ public class Workspace {
     }
 
     /// Updates the current dependencies.
-    public func updateDependencies() throws {
+    public func updateDependencies(repin: Bool = false) throws {
         let rootManifest = try loadRootManifest()
-        // Only create constraints based on root manifest for the update resolution.
-        let updateConstraints = computeRootPackageConstraints(rootManifest)
+        // Create constraints based on root manifest and pins for the update resolution.
+        let updateConstraints = computeRootPackageConstraints(rootManifest) 
+                              // Append the pin constraints if we don't have to repin.
+                              + (repin ? [] : pinsStore.createConstraints())
         // Resolve the dependencies.
         let updateResults = try resolveDependencies(constraints: updateConstraints)
+        // Update the checkouts based on new dependency resolution.
+        try updateCheckouts(with: updateResults)
+        // If we're repinning, update the pins store.
+        if repin {
+            for (specifier, version) in updateResults {
+                // FIXME: This is O(n), optimize.
```

Comment:
FWIW, I would probably write this the reverse way, where you go through every pin and update it if necessary. That makes it obvious that all current pins have been updated.